docs/pages/demo/datepicker/index.mdx

```diff
@@ -53,11 +53,11 @@ For that use `StaticDatePicker`.
   <Example paddingBottom source={StaticDatePicker} />
 </Hidden>
 
-#### Advanced keyboard input
+#### Custom input component
 
-We are providing default localized formats, but you can change this behaviour with `format` and `mask` props.
+You can customize rendering input by `renderInput` prop. But make sure you will spread `ref` and `inputProps` correctly to the custom input component.
```

Comment:
We must specify that props are https://material-ui.com/api/text-field/#textfield-api. Will refer this comment at #1761

Reply:
By the way the type of `renderInput` looks to awful

```tsx
(props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>
```

Because of core TextField typings (https://github.com/mui-org/material-ui/issues/20716)

Reply:
Would a similar approach to Autocomplete work?

```tsx
export interface RenderInputParams {
  id: string;
  disabled: boolean;
  InputLabelProps: object;
  InputProps: {
    ref: React.Ref<any>;
    className: string;
    startAdornment: React.ReactNode;
    endAdornment: React.ReactNode;
  };
  inputProps: object;
}
```